### PR TITLE
ci: Replace releaser with release please

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: .github/actions/ci/action.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+runs:
+  using: composite
+  steps:
+    - uses: haskell-actions/setup@v2
+      with:
+        ghc-version: ${{ inputs.ghc-version }}
+        enable-stack: true
+
+    # - name: Build library
```
Comment (kinyoklion) [inline .github/actions/ci/action.yml:21]: ?

File: .github/actions/publish/action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Setup dist directory
+      shell: bash
+      run: echo "STACK_DIR=$(stack --no-terminal path --dist-dir)" >> $GITHUB_ENV
+
+    - uses: haskell-actions/hackage-publish@v1
```
Comment (kinyoklion) [inline .github/actions/publish/action.yml:18]: It is nice that already has a dry run capability.

**Review by kinyoklion — APPROVED** [review]

File: .github/actions/ci/action.yml (hunk 1 above)
Comment (keelerm84) [reply .github/actions/ci/action.yml:21]: My bad. Cleaning that up.

File: .github/actions/publish/action.yml (hunk 2 above)
Comment (keelerm84) [reply .github/actions/publish/action.yml:18]: I know! This is the problem with working in so many different languages. None of them have everything you want!

**Pull request merged**